### DISCUSSION TITLE
Anyone having radio detection issue on RPI4 / docker with recent releases ?

### DISCUSSION THREAD

**Discussion by mulder999** [OpenCCU/OpenCCU#1883] [🇩🇪 Fragen&Antworten]
I am using raspberry pi 4 with RPI-RF-MOD connected on GPIO.
The setup has been fine up to raspmatic 3.61.7.20220226 on docker 20.10.14 on Linux aarch64 5.10.92-v7+

Then I started experiencing unreliable radio with raspmatic 3.63.8.20220330, so I had to revert to 3.61.7 to regain stable radio.

Yesterday I went with a system upgrade and now radio is no longer detected at all on docker 20.10.17 on Linux aarch64 5.15.32-v8+, no matter the raspmatic image.

In fact it ressembles some driver issue:

dmesg at boot:
`[    3.725544] eq3_char_loop: loading out-of-tree module taints kernel.`

Without much surprise radio detection fails at docker image start:
```
Identifying Homematic RF-Hardware: ...HmRF: n/a, HmIP: n/a, OK
Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found
```

Trying to isolate the problem one step at a time, but would love to see if anyone managed to run it with latest on everything on pi4.

PS: Still have a rpi3 with RPI-RF-MOD running on some older release/kernel and I confirmed both setups are indeed similar.

**Comment by heilerich** [comment]
I have the same setup: RPI-RF-MOD on GPIO

I can confirm that it is not working for me on the newest bullseye release (Kernel 5.15) as well. Same errors.

After a reinstall with buster (Kernel 5.10) it works as expected.

**Comment by mulder999** [comment]
@heilerich I finally managed to sort it. I forced reinstall of pivccu version 3.65.6-73 and had to mess around with `/boot/config.txt` to have
```
dtoverlay=disable-bt
```
instead of
<strike>
```
dtoverlay=miniuart-bt
enable_uart=1
```
</strike>

**Reply by heilerich** [reply]
And this is on the a host with a 5.15.x kernel?

**Reply by mulder999** [reply]
Yes this morning I was still on 5.15.32-v8+, after writing you I bumped up to 5.15.56-v8+ to control.

Here is more or less what I had to do after the kernel upgrade (still trying to figure out the minimal steps):
1. `sudo reboot` (otherwise it search the kernel headers at the former location)
2. `sudo apt reinstall pivccu3`
3. `sudo apt reinstall pivccu-modules-raspberrypi`
4. `echo "dtoverlay=pivccu-raspberrypi" | sudo tee /boot/config` (unless it is already there)
5. `sudo reboot`

First step is really key :)
Let me know if it helps.

REM:
I also upgraded raspberrymatic to latest, so far it works but seems a bit sluggish than before. Given the mess at/after every upgrade, I am really considering to freeze more and more. In French we call this super high fragility: `castle of cards`.
